cf/data/fragment/mixin/fragmentarraymixin.py

```diff
@@ -244,33 +166,12 @@ def _size_1_axis(self, indices):
         None
 
         """
-        axis = None
-
-        n_size_1 = 0  # Number of size 1 indices
-        for i, (index, n) in enumerate(zip(indices, self.shape)):
-            try:
-                x = index.indices(n)
-                if abs(x[1] - x[0]) == 1:
-                    # Index is a size 1 slice
-                    n_size_1 += 1
-                    axis = i
-            except AttributeError:
-                try:
-                    if index.size == 1:
-                        # Index is a size 1 numpy or dask array
-                        n_size_1 += 1
-                        axis = i
-                except AttributeError:
-                    if len(index) == 1:
-                        # Index is a size 1 list
-                        n_size_1 += 1
-                        axis = i
-
-        if n_size_1 > 1:
-            # There are two or more size 1 indices
-            axis = None
-
-        return axis
+        # REVIEW: getitem: `_size_1_axis`: refactor to use `original_shape`
+        original_shape = self.original_shape
+        if original_shape.count(1):
```

Comment:
> but also `original_shape` could be `None`

I don't see how, given that it is always returned as a `tuple` from the `IndexMixin.original_shape` method

Reply:
OK fair, I realise I misinterpreted the logic there a bit. Though at the moment, `original_shape` is erroring though because `shape` is not defined for the two new user-facing `Array` classes, as per one of my other comments (https://github.com/NCAS-CMS/cf-python/pull/805#discussion_r1807968144), so that needs sorting and then it should probably  good:

```pytb
>>> a = cf.data.array.NetCDF4Array()
>>> b = cf.data.array.H5netcdfArray()
>>> c = cf.data.array.mixin.IndexMixin()
>>> 
>>> a.original_shape
Traceback (most recent call last):
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 259, in _get_component
    return self._components[component]
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'shape'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/slb93/git-repos/cf-python/cf/data/array/mixin/indexmixin.py", line 362, in original_shape
    out = self.shape
          ^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/data/mixin/filearraymixin.py", line 52, in shape
    return self._get_component("shape")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 264, in _get_component
    return self._default(
           ^^^^^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 141, in _default
    raise default
ValueError: NetCDF4Array has no 'shape' component
>>> b.original_shape
Traceback (most recent call last):
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 259, in _get_component
    return self._components[component]
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'shape'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/slb93/git-repos/cf-python/cf/data/array/mixin/indexmixin.py", line 362, in original_shape
    out = self.shape
          ^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/data/mixin/filearraymixin.py", line 52, in shape
    return self._get_component("shape")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 264, in _get_component
    return self._default(
           ^^^^^^^^^^^^^^
  File "/home/slb93/git-repos/cfdm/cfdm/core/abstract/container.py", line 141, in _default
    raise default
ValueError: H5netcdfArray has no 'shape' component
>>> c.original_shape
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/slb93/git-repos/cf-python/cf/data/array/mixin/indexmixin.py", line 359, in original_shape
    out = self._custom.get("original_shape")
          ^^^^^^^^^^^^
AttributeError: 'IndexMixin' object has no attribute '_custom'
>>> c.shape
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'IndexMixin' object has no attribute 'shape'
```

Reply:
> so that needs sorting and then it should probably good

No need to sort this - it's just the expected error if the instance was not initialised with enough data.

Reply:
Ah OK yes I see - ignore me I was missing that `shape` is an `__init__` parameter to those `*Array` classes which is obvious now you point it out! Sorry for the confusing comments. Marking as resolved.